### ISSUE TITLE
Stand-alone package for Data.Massiv.Core.Index

### ISSUE THREAD

**Issue by dpwiz** [lehins/massiv#130]
I like how the library handles the size/index/etc. and would like to see more of that across the Hackage (instead of ad-hocs and/or copypasta).

How difficult it would be to put the Index module out as a stand-alone package?
Is there anything that prevents it?

**Comment by lehins** [comment]
I don't think it would be too difficult to extract all the index functionality into its own package. There isn't anything inherent that would prevent it.

I didn't think anyone would need it, but I can see how it could be useful outside of `massiv`. Maybe I'll extract it into its own package one of those days when I get some time on my hands
